### PR TITLE
feat: add deterministic tests for scaling of sink

### LINKED ISSUE #5856: Support scaling of `Sink`
Typically this should be trivial and covered in #5740 as the `Sink` is an alternative to `Materialize`. But do we also need to manage and reassign the "splits" for scaled sink nodes?

Any insights are welcome. 😄 cc @tabVersion @shanicky @nanderstabel

### PR DESCRIPTION
**Pull request by shanicky**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

fix #5856

Added simple tests for append_only and debezium sink

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/tests/simulation/tests/sink.rs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        })
+        .await;
+
+    let mut stream = consumer.stream();
+
+    for i in &[1, 2, 3, 4, 5] {
+        cluster
+            .run(&format!("insert into t values ({});", i))
+            .await?;
+        cluster.run(&format!("flush;")).await?;
```
Comment (BugenZhao) [inline src/tests/simulation/tests/sink.rs:182]:
```suggestion
        cluster.run("flush").await?;
```

File: src/tests/simulation/tests/sink.rs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+    let materialize_fragment = cluster
+        .locate_one_fragment([
+            identity_contains("materialize"),
+            no_identity_contains("globalSimpleAgg"),
+        ])
+        .await?;
+
+    let id = materialize_fragment.id();
+    cluster.reschedule(format!("{id}-[1,2,3,4,5]")).await?;
```
Comment (BugenZhao) [inline src/tests/simulation/tests/sink.rs:154]: What about also inserting and checking the sink records before scaling?

Comment (shanicky) [reply src/tests/simulation/tests/sink.rs:154]: make sense, i'll add some test

**Review by wangrunji0408 — COMMENTED** [review]
Generally LGTM! @BugenZhao Please help check the details. 🥰

**Review by BugenZhao — APPROVED** [review]
LGTM!

**Pull request merged**